### ISSUE TITLE
Chat colors don't update when changing transparent chat box setting

### ISSUE THREAD

**Issue by verafeiyd** [MarbleTurtle/Color-Coordinator#15]
Unfortunately, this one is very hard to reproduce, but over the year+ of using this plugin, ive noticed a rare occurrence where when typing / for clan chat, it will show the designated colour for friends chat. Now i assume this is probably a bug with slash swapper, as i do use it as well, but i figured it was worth mentioning.

I unfortunately dont know how to replicate it, but I do think it has to do with typing quick enough to get a character past the / before the check is made, which also.. makes it hard as a player to figure out if its even possible to fix but heres hoping.

love the plugin, fixes so much of my struggles with the chat. Thank you for your work

**Comment by ririshi** [comment]
Thanks for the compliment, happy to hear it works well for you. As for the bug, I haven't been playing (on RL) for a while so I don't know if it happens for me. Without a reproduction method, this will be hard to fix.

There are some situations where the color can change in quick succession when typing the first few characters, because of how chat messages are parsed by the native client code. I can't think of a specific example for your case off the top of my head, but it has something to do with needing or not needing a space after the slash(es).

Your guess about it being Slash Swapper sounds unlikely because the coloring is completely controlled by this plugin. The chat colors used by this plugin are set during initialization and again whenever a config setting related to chat colours or Slash Swapper is changed. So whenever you type `/` it should immediately change to the right color.

What you described would mean the chat color for 1 slash is set to friends chat color during initialization and the setting is not updated until after you start typing.

Perhaps you can do some testing to see if you can't reproduce it on purpose. Once there is a repro, I can make some time to find out what, if anything, is going wrong.

**Comment by verafeiyd** [comment]
Scratch that, I think I've found the problem. I've been using transparent chat for one of the first times in a while after removing a dark chatbox i use with my resource pack right, and since doing so, ive noticed when the game swaps from opaque to transparent or vice versa, it actually keeps the others ingame set colours for the new chatbox mode. ie. If i set public chat to black on opaque, the input colour most of the time (im testing by just swapping the chatbox on and off with text typed in the input), it ends up retaining it on transparent. 

heres a video/colour settings to showcase, this is ONLY while your plugin is active :sob:

![image](https://github.com/MarbleTurtle/Color-Coordinator/assets/123139321/f065d372-f586-4bf4-969d-2de8fc62d243)

https://i.gyazo.com/787adfe50999ee24c33ff16bb14e1c40.mp4
https://i.gyazo.com/79b6935ad4fe234994c7b86ca6d7066b.mp4

**Comment by verafeiyd** [comment]
i also leads to basically having opposite colours than what you want, weirdly

**Comment by ririshi** [comment]
The setting for transparent chatbox probably doesn't trigger the color mapping refresh. That would explain why you're seeing the wrong colors there. I can probably have a look and fix the issue when I find some free time.

**Comment by verafeiyd** [comment]
yeah, thats probably the case, but i will admit that it does happen on transparent chat whenever you have the normal chatbox pop up. (dialogue, level ups, etc). Thank you for looking into it though, cant stand talking to my clanmates in osrs without your plugin <3

**Comment by ririshi** [comment]
I finally found some time to start looking into this. I fixed the issue with the mapping not refreshing when you change from transparent to opaque, but I'm not sure what you mean by your last comment. Could you explain what you meant? Do your colours appear incorrectly after you close the normal chatbox?
